tests/apps/workflowsapp/app.go

```diff
@@ -0,0 +1,205 @@
+	log.Println("indexHandler is called")
+
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(appResponse{Message: "OK"})
+}
+
+// testHandler is the handler for end-to-end test entry point
+// test driver code call this endpoint to trigger the test
+func testHandler(w http.ResponseWriter, r *http.Request) {
+	// Retrieve request body contents
+	var commandBody testCommandRequest
+	err := json.NewDecoder(r.Body).Decode(&commandBody)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		json.NewEncoder(w).Encode(appResponse{
+			Message: err.Error(),
+		})
+		return
+	}
+
+	// Trigger the test
+	startTime := epoch()
+	statusCode, res := startTest(commandBody)
+	res.StartTime = startTime
+	res.EndTime = epoch()
+
+	w.WriteHeader(statusCode)
+	json.NewEncoder(w).Encode(res)
+}
+
+func startTest(commandRequest testCommandRequest) (int, appResponse) {
+	log.Printf("StartTest - message: %s", commandRequest.Message)
+
+	// Create json payload to send over HTTP to start workflow by providing task_queue
+	jsonData := []byte(`{
+		"workflow_options" :
+		{
+			"task_queue" : "e2e_test_queue"
+		}
+	}`)
+	res, err := httpClient.Post("http://localhost:3500/v1.0-alpha1/workflows/temporal/HelloTemporalWF/WorkflowID/start", "application/json", bytes.NewBuffer(jsonData))
+	if err != nil {
+		return http.StatusInternalServerError, appResponse{Message: err.Error()}
+	}
+	defer res.Body.Close()
+
+	// Get the data response from the start workflow call
+	body, _ := io.ReadAll(res.Body)
+	var resultData workflows.WorkflowReference
+	json.Unmarshal(body, &resultData)
+	time.Sleep(2 * time.Second)
+
+	// USe the data that was retrieved back from the start workflow call (InstanceID) to get info on the workflow
+	res, err = httpClient.Get("http://localhost:3500/v1.0-alpha1/workflows/temporal/HelloTemporalWF/" + resultData.InstanceID + "")
+	if err != nil {
+		return http.StatusInternalServerError, appResponse{Message: err.Error()}
+	}
+	defer res.Body.Close()
+
+	// Parse the data from the Get call, specifically status and return it. It should be complete
+	body, _ = io.ReadAll(res.Body)
+	var stateData workflows.StateResponse
+	json.Unmarshal(body, &stateData)
+	return http.StatusOK, appResponse{Message: stateData.Metadata["status"]}
+}
+
+// epoch returns the current unix epoch timestamp
+func epoch() int {
+	return int(time.Now().UnixMilli())
+}
+
+// appRouter initializes restful api router
+func appRouter() *mux.Router {
+	router := mux.NewRouter().StrictSlash(true)
+
+	// Log requests and their processing time
+	router.Use(utils.LoggerMiddleware)
+
+	router.HandleFunc("/", indexHandler).Methods("GET")
+	router.HandleFunc("/tests/{test}", testHandler).Methods("POST")
+
+	router.Use(mux.CORSMethodMiddleware(router))
+
+	return router
+}
+
+func main() {
+	log.Printf("Workflow Test - listening on http://localhost:%d", appPort)
+
+	client2, _ := client.NewNamespaceClient(client.Options{HostPort: "dapr-temporal-frontend.dapr-tests.svc.cluster.local:7233"})
```

Comment:
I think the address should be configurable via env var. The test will pass it as env vars, such as "SERVICE_NAME" and "NAMESPACE" and "PORT"... Don't just hardocde it!